tests/unit/store.spec.ts

```diff
@@ -701,6 +701,54 @@ describe('mutation tests', () => {
     expect(state.dataset).toEqual(dataset);
   });
 
+  it('sets rightDataset', () => {
+    buildState({
+      currentPipelineName: 'coco_l_asticot',
+      pipelines: {
+        coco_l_asticot: [],
+        dataset1: [],
+        dataset2: [],
+      },
+    });
+
+    const store = setupMockStore();
+    const dispatchSpy = jest.spyOn(store, 'dispatch');
+
+    store.dispatch(VQBnamespace('selectRightDataset'), { rightPipelineLabel: '' });
+    expect(dispatchSpy).not.toHaveBeenLastCalledWith(VQBnamespace('executePipeline'), '');
+
+    store.dispatch(VQBnamespace('selectRightDataset'), { rightPipelineLabel: 'coco_l_asticot' });
+    expect(dispatchSpy).not.toHaveBeenLastCalledWith(
+      VQBnamespace('executePipeline'),
+      [],
+      {
+        coco_l_asticot: [],
+        dataset1: [],
+        dataset2: [],
+      },
+      1,
+      0,
+    );
+
+    store.dispatch(VQBnamespace('selectRightDataset'), { rightPipelineLabel: 'other' });
+    expect(dispatchSpy).not.toHaveBeenLastCalledWith(
+      VQBnamespace('executePipeline'),
+      [
+        {
+          name: 'domain',
+          domain: 'other',
+        },
+      ],
+      {
+        coco_l_asticot: [],
+        dataset1: [],
+        dataset2: [],
+      },
+      1,
+      0,
+    );
+  });
```

Comment:
I think it'll be more useful to have test cases for the scenarios inside the `JoinStepForm`:
- when someone choose a right dataset, we should fetch the first row fo the selected right dataset and fill the `rightOn` options with it
- when someone changes the right dataset, we should update these options
- when someone edits an existing JoinStep, we should retrieve the right dataset right away

This may enlights some intermediate cases:
- do we need a spinner on the rightOn dropdown while we wait for the results?
- if we can't fetch the right dataset, should we default on a free input?
- what if someone selects a column, then changes dataset and this column doesn't exist in the newly selected dataset?

Reply:
Hi @davinov,

Thank you for your feedback.
Could you call me this afternoon ?

Have a nice day,

Charles

Reply:
Hi! Today is a bit packed and I'm off monday, but Tuesday could work!
Let's find a time together on slack :)